### PR TITLE
Save 0.13s (or more, it's hard to tell with Lavaland) by caching mineral spawn lists, and making effects not have integrity

### PR DESCRIPTION
**Pull request by Mothblocks**
Decals calling `getArmor` was 50ms, gives them `uses_integrity = FALSE` to counter this. I don't think this should have any visible effects.

Makes mineral spawn chances a proc that caches its results rather than a brand new list initialized on every single mineral (80+ms). Also calls `check_holidays` only once instead of over 30,000 times (which was 43.76ms). Also caches `smoothing_groups` and `canSmoothWith`. Numbers aren't wholly inaccurate Lavaland do be random.

### CODE REVIEW

File: code/game/turfs/closed/minerals.dm [hunk 1]
```diff
@@ -183,21 +188,34 @@
...
+	var/static/list/mineral_chances_by_type = list()
 
 	. = ..()
 	if (prob(mineralChance))
-		var/path = pick_weight(mineralSpawnChanceList)
+		var/list/spawn_chance_list = mineral_chances_by_type[type]
+		if (isnull(spawn_chance_list))
+			mineral_chances_by_type[type] = mineral_chances()
+			spawn_chance_list = mineral_chances_by_type[type]
+		var/path = pick_weight(spawn_chance_list)
```
Comment (LemonInTheDark) [inline code/game/turfs/closed/minerals.dm:218]: if you're going to cache this, may I suggest using expand_weights() and pick() rather then pick_weights()?
it has a memory cost but is significantly faster, and I bet would help here

Comment (LemonInTheDark) [reply code/game/turfs/closed/minerals.dm:218]: at least well, depending on how hot this actually is

Comment (Mothblocks) [reply code/game/turfs/closed/minerals.dm:223]: Alright

File: code/game/turfs/closed/minerals.dm [hunk 2]
```diff
@@ -39,6 +37,16 @@
...
 
+/turf/closed/mineral/Initialize(mapload)
+	var/static/list/smoothing_groups = list(SMOOTH_GROUP_CLOSED_TURFS, SMOOTH_GROUP_MINERAL_WALLS)
+	var/static/list/canSmoothWith = list(SMOOTH_GROUP_MINERAL_WALLS)
+
+	// The cost of the list() being in the type def is very large for something as common as minerals
+	src.smoothing_groups = smoothing_groups
+	src.canSmoothWith = canSmoothWith
+  
+  return ..()
```
Comment (Mothblocks) [inline code/game/turfs/closed/minerals.dm:48]: yeah i know hold on

**Pull request merged**